### ISSUE TITLE
'loading' page display when rendering template plotly_dash add embedded in.

### ISSUE THREAD

**Issue by fengjunCN** [GibbsConsulting/django-plotly-dash#126]
FYI verisons are
 dash 0.36.0
 dash-core-components 0.43.0
 dash-html-components 0.13.5
 django-plotly-dash 0.9.8

Installation of django-plotly-dash is ok;
 But when rendering the template, always stopped at loading dash app.
 template file:
 ......{%load plotly_dash%}
 {% plotly_app name = “SimpleExample”%}
 ......
So, what can i do next? Thanks!

-----------------
mysite/setting.py:
-----------------
INSTALLED_APPS = [
    'main.apps.MainConfig',
    'chart',
    'userAuth',
    'django_dash',
    'django_plotly_dash.apps.DjangoPlotlyDashConfig',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
]
MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django_plotly_dash.middleware.BaseMiddleware',
]
ROOT_URLCONF = 'mysite.urls'
TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]
WSGI_APPLICATION = 'mysite.wsgi.application'
# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}
# Password validation
# https://docs.djangoproject.com/en/2.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]
# Internationalization
# https://docs.djangoproject.com/en/2.1/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Shanghai'
USE_I18N = True
USE_L10N = True
USE_TZ = True
# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/2.1/howto/static-files/
STATIC_ROOT = '/var/www/mysite/static/'
STATIC_URL = '/static/'
-----------------------------
mysite/urls.py
----------------------------
from django.contrib import admin
#from django.urls import path
from django.urls import path, include
from django.conf.urls import url
# Use static() to add url mapping to serve static files during development (only)
from django.conf import settings
from django.conf.urls.static import static
from django_plotly_dash.views import add_to_session
urlpatterns = [
    path('',include('main.urls')),
    path('chart/',include('chart.urls')),
    url('^django_plotly_dash/', include('django_plotly_dash.urls')),
    path('django_dash/',include('django_dash.urls')),
    path('admin/', admin.site.urls),
    path('userAuth/', include('userAuth.urls')),
    #dash app
    #path('dash_within_django/', include('dash_within_django.urls')),

] 
------------------------
Django_dash/urls.py
------------------------
from django.views.generic import TemplateView
from django.urls import path, re_path, include
from django.conf.urls import url
from . import views
from . import plotly_apps 
from django_plotly_dash.views import add_to_session
app_name = 'django_dash'
urlpatterns = [
	url('^$', views.index_view, name="index_view"),
]

**Comment by stvbdn** [comment]
Having the same issue. Stuck on "loading..." even on the demo apps

**Comment by NeodarZ** [comment]
I had the same issue but I have successfully make this work. Ok it's not perfect but it's working.

I launch the following command:

```
./make_env
./prepare_redis
sed -i 's/-e git+https:\/\/github.com\/delsim\/django-plotly-dash.git@affccd8b280b2fdbc7f7e2d975b9cf4c7ace5f06#egg=django_plotly_dash/django_plotly_dash/g' frozen_dev.txt
sed -i 's/pkg-resources==0.0.0/#pkg-resources==0.0.0/g' frozen_dev.txt
source env/bin/activate
pip install -r frozen_dev.txt 
deactivate
./prepare_demo
```

It's seems to be a problem with some module not installed...

**Comment by NeodarZ** [comment]
Ok, after more investigations I found the problem. There is some issue with dash 0.39.0.

It seems there are no problem with the following package version:

```
dash==0.38.0
dash-core-components==0.43.1
dash-html-components==0.13.1
dash-renderer==0.19.0
```

@fengjunCN if I use the same version as you, all works for me... (I just tested with demo app, not with your code).

And the files `requirements.txt` and `dev_requirements.txt` must have a fixed version for dash to 0.36.0 for avoid 'Loading...' issue in django-plotly-dash 0.9.8.

**Comment by GibbsConsulting** [comment]
@NeodarZ Thanks for the observation and help. Created #127 - it looks like it could be a bit of work to move to 0.39

**Comment by fengjunCN** [comment]
> 
> 
> Ok, after more investigations I found the problem. There is some issue with dash 0.39.0.
> 
> It seems there are no problem with the following package version:
> 
> ```
> dash==0.38.0
> dash-core-components==0.43.1
> dash-html-components==0.13.1
> dash-renderer==0.19.0
> ```
> 
> @fengjunCN if I use the same version as you, all works for me... (I just tested with demo app, not with your code).
> 
> And the files `requirements.txt` and `dev_requirements.txt` must have a fixed version for dash to 0.36.0 for avoid 'Loading...' issue in django-plotly-dash 0.9.8.

yeap, it works now!  thanks!
